eclipse/plugins/org.switchyard.tools.ui/src/org/switchyard/tools/ui/M2EUtils.java

```diff
@@ -416,6 +420,110 @@ public static void hackIntegrationPackDependency(final Dependency dependency) {
         }
     }
 
+    /**
+     * Surf through the list of available artifact versions. 
+     * @param versions list incoming
+     * @return list outgoing
+     */
+    public static Set<ArtifactVersion> filterSwitchYardVersions(List<ArtifactVersion> versions) {
+        final String onePointZero = "1.0"; //$NON-NLS-1$
+        final Set<ArtifactVersion> filtered = new LinkedHashSet<ArtifactVersion>();
+        String previousMajorMinor = null;
+        boolean previousWasSnapshot = false;
+        /* assumes list is sorted lowest to highest. */
+        for (ListIterator<ArtifactVersion> it = versions.listIterator(versions.size()); it.hasPrevious();) {
+            ArtifactVersion next = it.previous();
+            String version = next.toString();
+            String majorMinor = getMajorMinor(version);
+            try {
+                if (previousMajorMinor == null) {
+                    filtered.add(next);
+                    previousWasSnapshot = version.endsWith("-SNAPSHOT"); //$NON-NLS-1$
+                } else if (version.endsWith("-SNAPSHOT")) { //$NON-NLS-1$
+                    // skip snapshots
+                    previousWasSnapshot = true;
+                    continue;
+                } else if (previousMajorMinor.equals(majorMinor) && !previousWasSnapshot) {
+                    // we already have this version
+                    continue;
+                } else {
+                    // don't have this one yet
+                    filtered.add(next);
+                    previousWasSnapshot = false;
+                    if (onePointZero.equals(majorMinor)) {
+                        // nothing before 1.0
+                        break;
+                    }
+                }
+            } finally {
+                previousMajorMinor = majorMinor;
+            }
+        }
+        return filtered;
+    }
+
+    /**
+     * Parse the whole version string to get the Major/Minor versions.
+     * @param version string
+     * @return string
+     */
+    public static String getMajorMinor(final String version) {
+        final int firstDot = version.indexOf('.');
+        final int secondDot = firstDot < 0 ? -1 : version.indexOf('.', firstDot + 1);
+        if (secondDot < 0) {
+            return version;
+        }
+        return version.substring(0, secondDot);
+    }
+    
+    /**
+     * Ensure that the default and previous SwitchYard versions show up in the available versions list.
+     * @param versions list incoming
+     * @return list outgoing
+     */
+    public static Set<ArtifactVersion> ensureDefaultAndPreviousVersionAvailable(final Set<ArtifactVersion> versions) {
+        // add default version
+        final ArtifactVersion defaultVersion = parseVersion(NewSwitchYardProjectWizard.DEFAULT_RUNTIME_VERSION);
+        if (!versions.contains(defaultVersion)) {
```

Comment:
Just realizing now: if it is a Set we don't need to check if it is contained inside first